esphome/core/application.cpp

```diff
@@ -108,6 +90,21 @@ void Application::loop() {
     this->components_[this->dump_config_at_]->call_dump_config();
     this->dump_config_at_++;
   }
+
+  const uint32_t elapsed = millis() - this->last_loop_;
+  uint32_t delay_time = 1;  // minimum delay due to tasks at prio 0
```

Comment:
```suggestion
  uint32_t delay_time = portTICK_PERIOD_MS;  // minimum delay due to tasks at prio 0
```

FreeRTOS schedules in "ticks", and the minimum possible delay is one tick, requesting a delay less than one tick is the same as `delay(0)`. This is how `delay()` is implemented.

```
void IRAM_ATTR HOT delay(uint32_t ms) { vTaskDelay(ms / portTICK_PERIOD_MS); }
```

Now I believe that in ESPHome by default `portTICK_PERIOD_MS` _is_ equal to 1, but using the defined constant eliminates any dependence on that.

Otherwise looks all good.

Reply:
Except that the constant is only available on esp-idf.

Adding a high delay kind of defeats the purpose of the PR.

I'll look into another way to trigger a schedule to a lower prio task. Or just attempt fix the current prio 0 tasks first, eventually that needs to be fixed anyway.

Reply:
> Except that the constant is only available on esp-idf.

What makes you think that? `portTICK_PERIOD_MS` is a constant associated with FreeRTOS which is used by ESPHome on all platforms, it is not specific to ESP-IDF.

Reply:
Now that the prio 0 task in components have been eliminated, I'll remove this patch from the PR.

How is FreeRTOS involved on archs other than esp32-idf? I don't see any task created and only esp32-idf has an `app_main()` function.

I hope my initial assumption that on other archs the "OS" or arch code takes care of other tasks between running the app's `loop()` function. If not please holler.

Reply:
Like I said "FreeRTOS ... is used by ESPHome on all platforms, it is not specific to ESP-IDF." 

> I hope my initial assumption that on other archs the "OS" or arch code takes care of other tasks between running the app's `loop()` function. If not please holler.

Assumptions are often wrong. The best venue for questions like that is in the Discord channels, probably [#off-topic](https://discord.com/channels/429907082951524364/822162083452813312).